src/parser/parser_support.c

```diff
@@ -10968,39 +10968,198 @@ pt_get_server_name_list (PARSER_CONTEXT * parser, PT_NODE * node, void *arg, int
   return node;
 }
 
+static char *
+skip_comment (char *ps)
+{
+  while (*ps)
+    {
+      if (ps[0] == '*' && ps[1] == '/')
+	{
+	  ps += 2;
+	  break;
+	}
+      ps++;
+    }
+  return ps;
+}
+
+char *
+read_id (char *ps, char end_char)
+{
+  while (*ps != end_char)
+    {
+      ps++;
+    }
+
+  return ((*ps == end_char) ? (ps + 1) : ps);
+
+}
+
+static char *
+read_string (char *ps, char end_char, bool no_escape)
+{
+  if (no_escape)
+    {
+      while (*ps)
+	{
+	  if (ps[0] == end_char)
+	    {
+	      if (ps[1] != end_char)
+		{
+		  return ps + 1;
+		}
+	      ps++;
+	    }
+	  ps++;
+	}
+    }
+  else
+    {
+      while (*ps)
+	{
+	  if (ps[0] == end_char)
+	    {
+	      if (ps[1] != end_char)
+		{
+		  return ps + 1;
+		}
+	      ps++;
+	    }
+
+	  else if (ps[0] == '\\')
+	    {
+	      switch (ps[1])
+		{
+		case '\'':
+		case '"':
+		case 'n':
+		case 'r':
+		case 't':
+		case '\\':
+		case '%':
+		case '_':
+		  ps++;
+		  break;
+
+		default:
+		  break;
+		}
+	    }
+	  ps++;
+	}
+    }
+
+  return ps;
+}
+
+static char *
+find_circle_at_char (bool ansi_quotes, bool no_escape, char *ps)
```

Comment:
The reason for using sql_user_text is as follows.
1) Here we need the user's original query that has not been rewritten in the middle.
2) It is not easy to get the top root node of the tree.

Reply:
This function will not be used, because I change the routine to use parser_print_tree or pt_print_bytes to add the custom print flag "PT_SUPPRESS_SERVER_NAME".